### DISCUSSION TITLE
How to load a function written in a .r file into environment while rendering a Quarto Project?

### DISCUSSION THREAD

**Discussion by ming-cui** [quarto-dev/quarto-cli#6063] [Q&A] [answered]
### Description

For the individual `.qmd` files, the code blocks use a large function written in a `.r` file. I will first open the `.r` file and run all the code in it to make `.qmd` files work. Specifically, for the code below, I need to first open the `.r` file, execute all the code in it, and then get a function called `A_function_written_in_dotR_file`. So, rendering the project containing the code below will fail because I couldn't find a way to automatically load the function `A_function_written_in_dotR_file` into the environment. 

````qmd
---
title: "Reproducible Quarto Document"
format: html
---

```{r}
A_function_written_in_dotR_file(cars)
```

The end.
````

I got this error message when trying to render the project:

```
21% (unnamed-chunk-3) Quitting from lines 107-123 (sample.qmd) 
Error in file(file, "rt") : cannot open the connection
Calls: .main ... eval_with_user_handlers -> eval -> eval -> read.table -> file
```

**Comment by mcanouil** [comment]
Could you edit your post with both a question in the title and in the body?

Also, could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by mcanouil** [comment, marked as answer]
Why don't you use `source()`?

**Reply by cderv** [reply]
Additional resource on this: 

This is the recipe for **knitr** on how to source an external script in a Project. https://bookdown.org/yihui/rmarkdown-cookbook/source-script.html

This works with Quarto too.
